### ISSUE TITLE
reconnectToPeer method does not stop.

### ISSUE THREAD

**Issue by qkrwnsgh1288** [tendermint/tendermint#2894]
Let's say that the situation is node1 and persistentNode are connected, and then persistentNode is stopped, and then persistentNode is restarted.

node1 calls the reconnectToPeer goroutine to establish a connection with the persistentPeer.
persistentPeer knows the address of node1, so it makes an outbound connection to node1 for reconnection.
node1 accepts the connection request of the persistentPeer and connects to each other.
However, the reconnectToPeer goroutine does not know this and continues to send a connection request to the persistentPeer.
So, when the persistentPeer receives the request, it continually exposes an error that is already registered as a peer.

To summarize,
When the persistentPeer sends a connection request to the node1 first, the reconnectToPeer goroutine, which is driven by the peer side, does not know that the connection is established and sends a connection request continuously.

This seems to be a waste of resources.

**Comment by ebuchman** [comment]
Thanks - believe this is a duplicate of https://github.com/tendermint/tendermint/issues/2716
